### ISSUE TITLE
Accept tidyverse tilde functions

### ISSUE THREAD

**Issue by davidsjoberg** [tidyverse/ggplot2#4103]
It would be neat to be able to use the same "short" version of defining function as in `purrr` in `ggplot2` as well.

---

Example would be to be able to write:

```r
ggplot(mtcars, aes(drat, qsec)) + 
  geom_line() +
  scale_y_continuous(labels = ~format(., nsmall = 1))
```

Today you need to do like this:

```r
ggplot(mtcars, aes(drat, qsec)) + 
  geom_line() +
  scale_y_continuous(labels = function(x) format(x, nsmall = 1))
```

Thanks! :)

**Comment by yutannihilation** [comment]
Yes, I think all the arguments that accepts functions should accept tilde notation. It's just we don't have enough development bandwidth to do it. Pull requests are welcome!

**Comment by davidsjoberg** [comment]
Which parameters accept functions?

**Comment by yutannihilation** [comment]
This now works. Probably fixed by #4188.

``` r
library(ggplot2)

ggplot(mtcars, aes(drat, qsec)) + 
  geom_line() +
  scale_y_continuous(labels = ~format(., nsmall = 1))
```

![](https://i.imgur.com/HO7TCpx.png)

<sup>Created on 2022-07-29 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>
